internal/providers/pluginfw/resources/sharing/data_shares.go

```diff
@@ -0,0 +1,67 @@
+package sharing
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-framework/types"
+
+	"github.com/databricks/databricks-sdk-go/service/sharing"
+	"github.com/databricks/terraform-provider-databricks/common"
+	pluginfwcommon "github.com/databricks/terraform-provider-databricks/internal/providers/pluginfw/common"
+	"github.com/databricks/terraform-provider-databricks/internal/providers/pluginfw/tfschema"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+)
+
+type SharesList struct {
+	Shares []types.String `tfsdk:"shares" tf:"computed,optional,slice_set"`
+}
+
+func DataSourceShares() datasource.DataSource {
+	return &SharesDataSource{}
+}
+
+var _ datasource.DataSourceWithConfigure = &SharesDataSource{}
+
+type SharesDataSource struct {
+	Client *common.DatabricksClient
+}
+
+func (d *SharesDataSource) Metadata(ctx context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = pluginfwcommon.GetDatabricksStagingName("shares")
```

Comment:
Same here

Reply:
Initially, I planned to use a `dataSourceName` constant for both the _share_ and _shares_ data sources. However, since they share the same namespace, that wasn’t an option. So, I just included the constant within the `Metadata` function, particularly since it’s only used once in each data source. Alternatively, I could assign a `dataSourceName` constant to one data source and rename it for the other, but that didn't feel quite right.

Let me know if you have a preferred approach.

Reply:
I see, yeah we could use dataSourceNameShare, dataSourceNameShares, mentioning this because we would also need to set the useragent (eg: https://github.com/databricks/terraform-provider-databricks/blob/main/internal/providers/pluginfw/resources/volume/data_volumes.go#L56) and use the same name there as well so might be better to abstract this in a single variable.